### PR TITLE
Development

### PR DESCRIPTION
**Pull request by josephhalstead**
Quite a few changes here:

1) bin/calculate_sensitivity.sh - change so that we recalculate coverage on GIAB sample before calculating sensitivity. Due to the fact that the original coverage calculation BED file has changed slightly (see coverage)

2) bin/calculate_sex.py - changed to unify with FH pipeline

3) bin/create_depth_summary.py - new script to calculate coverage summary for each sample

4) bin/germline_variant_reporter.py - already reviewed in dragenwgs_post_processing. Not part of validated test. No review reequired.

5) bin/get_coverage_region_stats.py - replace old bash based script with python - same as FH pipeline now. Requires review.

6) bin/make_database_config.py - config file generator for new variant database. Already in dragenwgs_post_processing - no review needed.

7) bin/quality_filter.sh - new dragen (3.7) has new quality filtering characteristics. Allowed us to simpligy quality filterting for SNP/Indels/

8) config/IlluminaTruSightOne/IlluminaTruSightOne_wren.config - new config for WREN. New quality filtering values

9) dragenge_post_processing.nf - new annotations for VCF.

Overall the changes made are to support pipeline execution on WREN, support new Dragen version 3.7 and support new coverage calculations.

### CODE REVIEW

File: bin/germline_variant_reporter.py [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+python germline_variant_reporter.py \
+    --vcf $vcf \
+    --proband_id ${sample_names[0]} \
+    --ped $ped \
+    --gene_list $gene_panel \
+    --gnomad_ad $gnomad_ad \
+    --gnomad_r $gnomad_r \
+    --minqual_snps $snp_qual \
+    --minqual_indels $indel_qual \
+    --min_dp $min_dp \
```
Comment (christianfiles) [inline bin/germline_variant_reporter.py:28]: $min_dp and $min_gq maybe rename to $min_dp_variant_calling and $min_gq_variant_calling to make more clear?

File: dragenge_post_processing.nf [hunk 2]
```diff
@@ -36,422 +37,536 @@ Define initial channels
...
+    publishDir "${params.publish_dir}/ped/", mode: 'copy'
 
-	input:
-    set val(id), file(vcf), file(vcf_index) from raw_vcf 
+    input:
+    file(variables) from variables_ped_ch.collect()
 
     output:
-    set file("${params.sequencing_run}.roi.vcf.gz"), file("${params.sequencing_run}.roi.vcf.gz.tbi") into roi_vcf_channel
+    file("${params.sequencing_run}.ped") into ped_ch
```
Comment (christianfiles) [inline dragenge_post_processing.nf:90]: Maybe create multiple channels here to make code look more tidy?
E.g. file("${params.sequencing_run}.ped") into ped_config_ch, ped_reporting_ch

File: dragenge_post_processing.nf [hunk 3]
```diff
@@ -36,422 +37,536 @@ Define initial channels
...
-	bcftools view -R $capture_bed $vcf > ${params.sequencing_run}.roi.vcf
+    """
+    create_ped.py --variables "*.variables" > ${params.sequencing_run}.ped
+    """
+}
 
-	bgzip ${params.sequencing_run}.roi.vcf
-	tabix ${params.sequencing_run}.roi.vcf.gz
-	"""
+ped_ch.into{
```
Comment (christianfiles) [inline dragenge_post_processing.nf:97]: See above comment. Can get rid of this.

File: dragenge_post_processing.nf [hunk 4]
```diff
@@ -36,422 +37,536 @@ Define initial channels
...
 
-	input:
-	set file(vcf), file(vcf_index) from vcf_annotation_channel 
+    input:
+    set file(vcf), file(vcf_idx) from roi_vcf_ch
 
-	output:
-	set file("${params.sequencing_run}.roi.filtered.norm.vcf.gz"), file("${params.sequencing_run}.roi.filtered.norm.vcf.gz.tbi") into normalised_vcf_channel
+    output:
+    set file("${params.sequencing_run}_roi_qual.vcf.gz"), file("${params.sequencing_run}_roi_qual.vcf.gz.tbi") into quality_filtered_vcf_ch
```
Comment (christianfiles) [inline dragenge_post_processing.nf:159]: Same as above with multiple channels. Separate with ',' to create multiple channels

File: dragenge_post_processing.nf [hunk 5]
```diff
@@ -36,422 +37,536 @@ Define initial channels
...
+    $params.indel_qual \
+    $params.reference_genome \
+    "$params.medium_java_options" \
+    $params.min_dp_variant_reporting
+    """
 
-	"""
+}
 
+quality_filtered_vcf_ch.into{
```
Comment (christianfiles) [inline dragenge_post_processing.nf:174]: Can get rid of this if above comment is applied

File: dragenge_post_processing.nf [hunk 6]
```diff
@@ -36,422 +37,536 @@ Define initial channels
...
 
     publishDir "${params.publish_dir}/annotated_vcf/", mode: 'copy'
 
     input:
-    file(normalised_vcf) from normalised_vcf_channel
+    set file(quality_vcf), file(quality_vcf_idx) from quality_filtered_vcf_annotation_ch
 
     output:
-    set file("${params.sequencing_run}.roi.filtered.norm.anno.vcf.gz"), file("${params.sequencing_run}.roi.filtered.norm.anno.vcf.gz.tbi")  into annotated_vcf
+    set file("${params.sequencing_run}_anno.vcf.gz"), file("${params.sequencing_run}_anno.vcf.gz.tbi") into annotated_vcf_ch
```
Comment (christianfiles) [inline dragenge_post_processing.nf:193]: Create multiple channels here

File: dragenge_post_processing.nf [hunk 7]
```diff
@@ -36,422 +37,536 @@ Define initial channels
...
-    tabix ${params.sequencing_run}.roi.filtered.norm.anno.vcf.gz
+    slivar expr --gnotate $gnotate_gnomad -o ${params.sequencing_run}_anno.vcf -v ${params.sequencing_run}_vep.spliceai.vcf.gz
+    bgzip ${params.sequencing_run}_anno.vcf
+    tabix ${params.sequencing_run}_anno.vcf.gz
 
+    md5sum ${params.sequencing_run}_anno.vcf.gz > ${params.sequencing_run}.norm.anno.vcf.gz.md5
     """
 }
 
+annotated_vcf_ch.into{
```
Comment (christianfiles) [inline dragenge_post_processing.nf:252]: Get rid of this

File: dragenge_post_processing.nf [hunk 8]
```diff
@@ -36,422 +37,536 @@ Define initial channels
...
 
-	input:
-    set val(id), file(bam), file(bam_index) from coverage_bams 
+    input:
+    set val(id), file(bam), file(bam_index) from original_bams_coverage_ch
 
     output:
-    set val(id), file("${id}_depth_of_coverage.gz"), file("${id}_depth_of_coverage.gz.tbi") into per_base_coverage_channel
-    file("${id}_depth_of_coverage.sample_summary") into depth_summary
+    set val(id), file("${id}_depth_of_coverage.csv.gz"), file("${id}_depth_of_coverage.csv.gz.tbi") into per_base_coverage_ch
```
Comment (christianfiles) [inline dragenge_post_processing.nf:437]: Create multiple channels here

File: dragenge_post_processing.nf [hunk 9]
```diff
@@ -36,422 +37,536 @@ Define initial channels
...
+    """
 
-// split filtered vcf channel into four
-per_base_coverage_channel.into{
-    per_base_coverage_channel_gaps
-    per_base_coverage_channel_sex
-    per_base_coverage_channel_sensitivity
 }
 
+per_base_coverage_ch.into{
```
Comment (christianfiles) [inline dragenge_post_processing.nf:462]: Get rid of this

**Comment by christianfiles** [comment]
Changes requested

**Comment by josephhalstead** [comment]
made changes as requested. Running 191010_D00501_0366_BH5JWHBCX3 on WREN to confirm no accidental changes to performance

**Comment by josephhalstead** [comment]
checked and merged

**Pull request merged**